x-pack/test/saved_object_api_integration/common/suites/import.ts

```diff
@@ -8,23 +8,34 @@
 import expect from '@kbn/expect';
 import { SuperTest } from 'supertest';
 import type { Client } from '@elastic/elasticsearch';
+import type { SavedObjectReference } from 'src/core/server';
 import { SAVED_OBJECT_TEST_CASES as CASES } from '../lib/saved_object_test_cases';
 import { SPACES } from '../lib/spaces';
 import { expectResponses, getUrlPrefix, getTestTitle } from '../lib/saved_object_test_utils';
 import { ExpectResponseBody, TestCase, TestDefinition, TestSuite } from '../lib/types';
 
 export interface ImportTestDefinition extends TestDefinition {
```

Comment:
The `import` and `resolveImportErrors` integration tests were designed to import a mixture of objects that don't exist yet, with objects that already exist. This would generate conflict errors, and the destination objects would need to be overwritten. Each suite exercises three different scenarios:

1. `createNewCopies: false, overwrite: false`
   ![image](https://user-images.githubusercontent.com/5295965/148425718-0d14db01-ceec-4295-9e19-20fb42142daf.png)
2. `createNewCopies: false, overwrite: true`
   ![image](https://user-images.githubusercontent.com/5295965/148425776-4911a50a-057f-4c52-b201-9fb54a399c3c.png)
3. `createNewCopies: true`, overwrite: false`
   ![image](https://user-images.githubusercontent.com/5295965/148425856-b73d3bd4-2762-49ff-9ae0-73b51c6016ce.png)

The problem is there is another type of error that can occur, a "missing references" error, that originates from the `validateReferences` module. We never had tests for this before, and I thought it would be prudent to add tests for this now, since we need to make assertions that Kibana is correctly matching origins for references anyway.

Unfortunately, these test suites needed some significant refactoring to make this happen. There are a lot of ways to skin this cat, but I tried to make the most minimal changes possible. I think there is a lot of room for improvement, but I didn't want this PR to be any bigger than it already is.